Sami/Tests/Parser/NodeVisitorTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Sami\Tests\Parser;
+
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Name\Relative;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\NodeTraverser;
+use PhpParser\PrettyPrinter\Standard;
+use Sami\Parser\DocBlockParser;
+use Sami\Parser\Filter\TrueFilter;
+use Sami\Parser\NodeVisitor;
+use Sami\Parser\ParserContext;
+use Sami\Project;
+use Sami\Reflection\ClassReflection;
+use Sami\Reflection\MethodReflection;
+use Sami\Reflection\ParameterReflection;
+use Sami\Store\ArrayStore;
+
+/**
+ * Test of node visitor.
+ *
+ * @author Tomasz Struczyński <t.struczynski@gmail.com>
+ */
+class NodeVisitorTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Test method type hints.
+     *
+     * @dataProvider methodTypehintsDataProvider
```

Comment:
I'd recommend sticking with some common prefix/suffix in data provider method names so that we can easily see while browsing class source code which of the methods are data providers for somebody.